### PR TITLE
Refactor allthethings

### PR DESCRIPTION
**Pull request by inkel**
Big refactor of everything:
- ditch the spec syntax
- remove [`micromachine`](https://github.com/soveran/micromachine) dependency
- reduce public API exposure
- disable defaults

### CODE REVIEW

**Comment by inkel** [comment]
So Ruby 2.1 and Rubinius fails. Both fail with a `NameError` for `MiniTest::Test`.

File: test/disyuntor_test.rb [hunk 1]
```diff
@@ -0,0 +1,205 @@
+if ENV["COVERAGE"]
+  require "simplecov"
+  SimpleCov.start do
+    add_filter "/test/"
+    add_filter "/spec/"
```
Comment (fsaravia) [inline test/disyuntor_test.rb:5]: Given that you removed the `spec` folder, maybe you can remove this filter as well

File: test/disyuntor_test.rb [hunk 2]
```diff
@@ -0,0 +1,205 @@
...
+      end
+    end
+
+    assert_equal 42, @disyuntor.try { fail CustomRuntimeError }
+  end
+
+  def test_count_failures
+    assert_equal 0, @disyuntor.failures
+
+    @disyuntor.threshold.times do
```
Comment (fsaravia) [inline test/disyuntor_test.rb]: What do you think about testing a different number than `threshold` here? Just for the sake of testing a different scenario than the one that opens the circuit.

**Comment by inkel** [comment]
@fsaravia care to take another look?

**Comment by fsaravia** [comment]
@inkel loos great! :)

**Pull request merged**